client/src/utils/processImageHtml.ts

```diff
@@ -4,7 +4,19 @@ export function processImageHtml(html: string): string {
 
   return html.replace(/<img\s+([^>]*?)src="([^"]+)"([^>]*?)>/g, (match, before, src, after) => {
     try {
-      const replacedSrc = src.replace(s3Domain, cloudfrontDomain);
+      let replacedSrc = src.replace(s3Domain, cloudfrontDomain);
+
+      const url = new URL(replacedSrc);
+      const pathParts = url.pathname.split('/');
+      const filename = pathParts[pathParts.length - 1];
+
+      const hasExtension = /\.(jpeg|jpg|png|gif|webp)$/i.test(filename);
+      const processedFilename = hasExtension ? filename : `${filename.replace(/\.[^/.]+$/, '')}.jpeg`;
+
+      pathParts[pathParts.length - 1] = processedFilename;
+      url.pathname = pathParts.join('/');
+
+      replacedSrc = url.toString();
```

Comment:
```suggestion
      const replacedSrc = src.replace(s3Domain, cloudfrontDomain);

      const url = new URL(replacedSrc);
      const pathParts = url.pathname.split('/');
      const filename = pathParts[pathParts.length - 1];

      const hasExtension = /\.(jpeg|jpg|png|gif|webp)$/i.test(filename);
      const processedFilename = hasExtension ? filename : `${filename.replace(/\.[^/.]+$/, '')}.jpeg`;

      pathParts[pathParts.length - 1] = processedFilename;
      url.pathname = pathParts.join('/');

      const processedSrc = url.toString();
```

replacedSrc를 const로 선언하는 것을 고려해볼 수 있을 것 같아요!